### ISSUE TITLE
Validation with sacrebleu tokenizes escaped UNK-tokens

### ISSUE THREAD

**Issue by villmow** [pytorch/fairseq#1827]
## 🐛 Bug

I'm on a translation task and do validation with BLEU during training. If `--eval-tokenized-bleu` is not set, scoring is done using the sacrebleu.DEFAULT_TOKENIZER. See these lines:
https://github.com/pytorch/fairseq/blob/d7b0bf7e55b48466ee7e371a959668dfb8feaaf6/fairseq/tasks/translation.py#L348-L357

UNK-tokens in references are escaped with `<<UNK>>` before scoring (line 351). This may produce a reference like this `This is a new <<UNK>>.`. In hypos the UNK-tokens are not escaped, leaving a hypothesis like this `This is a new <UNK>.`

**Bug:** The default tokenizer of sacrebleu breaks the UNK-tokens (which is probably not wanted behaviour) and scores each token separately, altering the bleu scores.

### To Reproduce

Follow any translation example.

#### Code sample
This is whats currently implemented:
```python
>>> hypo = "This is a new <UNK>."
>>> ref = "This is a new <<UNK>>."
>>> from sacrebleu import TOKENIZERS, DEFAULT_TOKENIZER
>>> TOKENIZERS[DEFAULT_TOKENIZER](hypo)
'This is a new < UNK > .'
>>> TOKENIZERS[DEFAULT_TOKENIZER](ref)
'This is a new < < UNK > > .'
>>> sacrebleu.corpus_bleu([hypo], [[ref]], tokenize=DEFAULT_TOKENIZER).score
55.96526475152516
```

This is how I think these sequences should be scored:
```python
>>> hypo = "This is a new ABC."
>>> ref = "This is a new UNK."
>>> TOKENIZERS[DEFAULT_TOKENIZER](ref)
'This is a new UNK .'
>>> TOKENIZERS[DEFAULT_TOKENIZER](hypo)
'This is a new ABC .'
>>> sacrebleu.corpus_bleu([hypo], [[ref]], tokenize=DEFAULT_TOKENIZER).score
53.7284965911771
```
### Expected behavior

The bleu score should not be influenced by the unknown token.

### Environment

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.2
 - OS (e.g., Linux): linux
 - How you installed fairseq (`pip`, source): source

**Comment by myleott** [comment]
Fixed by b2ee110c853c5effdd8d21f50a8437485bafb285

**Comment by sukannyapurkayastha** [comment]
I also see that after fairseq.preprocess, some tokens are not there in the dicts for both the languages. What might be the reason?
